packages/web-components/src/components/fade-in-out/fade-in-out.ts

```diff
@@ -0,0 +1,256 @@
+  keepAnimation = false;
+
+  /**
+   * Amount of columns used for calculation.
+   *
+   * @private
+   */
+
+  private _colSpan = 3;
+
+  /**
+   * Saved list of elements to observe to avoid calling querySelectorAll
+   * more than once.
+   *
+   * @private
+   */
+  private _elements: HTMLElement[] = [];
+
+  /**
+   * Intersection Observer that watches outer viewport.
+   *
+   * @private
+   */
+  private _rootObserver: IntersectionObserver | null = null;
+
+  /**
+   * Intersection observer that watches the inner viewport.
+   *
+   * @private
+   */
+  private _innerObserver: IntersectionObserver | null = null;
+
+  /**
+   * Resize observer to trigger rootMargin recalculations
+   *
+   * @private
+   */
+  private _resizeObserver: any | null = null;
+
+  /**
+   * Intersection Observer options
+   *
+   * @private
+   */
+  private _options = {
+    rootMargin: '0px',
+    threshold: 0,
+  };
+
+  /**
+   * The inner viewport calculation for the root margins.
+   *
+   * @private
+   */
+  private _getViewportMargin() {
+    return `-${((document.documentElement.clientHeight * this._colSpan) / breakpoints.max.columns).toString()}px 0px`;
+  }
+
+  /**
+   * Cleans observers upon update or exit, and creates new instances if needed.
+   *
+   * @param {boolean} create to create new observer instances
+   */
+  private _cleanAndCreateObservers({ create }: { create?: boolean } = {}) {
+    this._cleanAndCreateRootObserver();
+    this._cleanAndCreateInnerObserver();
+    this._cleanAndCreateResizeObserver();
+
+    if (create) {
+      this._cleanAndCreateRootObserver({ create: true });
+      this._cleanAndCreateResizeObserver({ create: true });
+
+      const { selectorTargets } = this;
+      if (selectorTargets) {
+        forEach(document.querySelectorAll(selectorTargets), item => {
+          this._rootObserver?.observe(item);
+          this._elements.push(item as HTMLElement);
+        });
+      }
+      this._resizeObserver.observe(document.documentElement);
+    }
+  }
+
+  /**
+   * Cleans-up and creats the root intersection observer.
+   *
+   * @param [options] The options.
+   * @param [options.create] `true` to create the new intersection observer.
+   */
+  private _cleanAndCreateRootObserver({ create }: { create?: boolean } = {}) {
+    if (this._rootObserver) {
+      this._rootObserver.disconnect();
+      this._rootObserver = null;
+    }
+
+    if (create) {
+      this._rootObserver = new IntersectionObserver(this._handleExit);
+    }
+  }
+
+  /**
+   * Cleans-up and creats the inner intersection observer.
+   *
+   * @param [options] The options.
+   * @param [options.create] `true` to create the new intersection observer.
+   */
+  private _cleanAndCreateInnerObserver({ create }: { create?: boolean } = {}) {
+    if (this._innerObserver) {
+      this._innerObserver.disconnect();
+      this._innerObserver = null;
+    }
+
+    if (create && this._elements) {
```

Comment:
Instead of creating `this._elements` in another function, we can run `qSA` with `selectorTarget` here.